src/store/types.ts

```diff
@@ -0,0 +1,17 @@
+export interface Post {
+    id: string;
+    image: string;
+    selectedTags: string[];
```

Comment:
There's a bit of confusion with `selectedTags` and `tags` in your code. I think you should use `tags` throughout.